### PR TITLE
Fix: Cannot add multiple attribute decorators for primary key

### PR DESCRIPTION
**Pull request by blazejkustra**
#### Model
```ts
import attribute from '../dist/decorators';
import Entity from '../dist/entity';
import TableManager from '../dist/table';

export class TableUser extends Entity {
  @attribute.gsi.partitionKey.string({ indexName: 'index-1' })
  @attribute.partitionKey.string()
  domain: string;

  @attribute.sortKey.string()
  @attribute.gsi.sortKey.string({ indexName: 'index-1' })
  email: string;

  constructor(data: { domain: string; email: string }) {
    super(data);
    this.domain = data.domain;
    this.email = data.email;
  }
}

export const UserdataTableManager = new TableManager(TableUser, {
  tableName: 'USERS_TABLE_NAME',
  partitionKey: 'domain',
  sortKey: 'email',
  indexes: {
    'index-1': {
      partitionKey: 'domain',
      sortKey: 'email',
    },
  },
});

const EntityManager = UserdataTableManager.entityManager();
```

#### General
```ts
async function test() {
  const entity = await EntityManager.put(
    new TableUser({
      domain: 'test',
      email: 'not_empty',
    }),
  );
  console.log(entity);
}

async function createTable() {
  const table = await UserdataTableManager.createTable();
  console.log(table);
}

createTable();
test();
```

### GitHub linked issue:

#30 

### Type of change:
- [x] Bug fix
- [ ] Feature implementation
- [ ] Documentation improvement
- [ ] Testing improvement
- [ ] Something not listed here

### Is this a breaking change?
- [ ] YES 🚨
- [x] No

### Other:
- [x] I have searched through the GitHub pull requests to ensure this PR has not already been submitted
- [x] I have updated the Dynamode documentation (if required)
- [x] I have added/updated the Dynamode test cases (if required) 
- [x] I agree that all changes made in this pull request may be distributed and are made available in accordance with the [Dynamode License](../LICENSE).

### CODE REVIEW

**Comment by blazejkustra** [comment]
In case you have any questions fell free to leave them here @afronski 👍 

I'll add some e2e tests next week.

File: lib/dynamode/storage/index.ts [hunk 1]
```diff
@@ -138,28 +159,25 @@ export default class DynamodeStorage {
...
-    validateMetadataUniqueness(entityName, metadata);
-
     // Validate decorated attributes
     Object.entries(attributes).forEach(([name, attribute]) =>
       validateDecoratedAttribute({ name, attribute, metadata, entityName }),
     );
 
     // Validate table partition key
-    validateMetadataAttribute({ name: metadata.partitionKey, attributes, role: 'partitionKey', entityName });
+    validateMetadataAttribute({ name: metadata.partitionKey, attributes, validRoles: ['partitionKey'], entityName });
```
Comment (afronski) [inline lib/dynamode/storage/index.ts:168]: Why do you need to enforce that? 

It shouldn't be the case - e.g., you can use the main table *sort key* as a GSI *partition key*.

File: lib/dynamode/storage/index.ts [hunk 2]
```diff
@@ -138,28 +159,25 @@ export default class DynamodeStorage {
...
 
     // Validate table sort key
     if (metadata.sortKey) {
-      validateMetadataAttribute({ name: metadata.sortKey, attributes, role: 'sortKey', entityName });
+      validateMetadataAttribute({ name: metadata.sortKey, attributes, validRoles: ['sortKey'], entityName });
```
Comment (afronski) [inline lib/dynamode/storage/index.ts:172]: As in case for `partitionKey` check.

**Review by afronski — CHANGES_REQUESTED** [review]
One remark: as in the proposed example, the original case from #30 looks good.

However, one of the allowed scenarios does not work:

```typescript
export class TableUser extends Entity {
  @attribute.partitionKey.string()
  id: string;

  // --> This throws a "role mismatch" - and it's perfectly allowed as a reprojection for GSI. <--
  // "Attribute "type" is decorated with a wrong role in "TableUser" Entity."
  @attribute.sortKey.string()
  @attribute.gsi.partitionKey.string({ indexName: 'index-1' })
  type: string;

  @attribute.gsi.sortKey.string({ indexName: 'index-1' })
  email: string;
  
  // ...
}
```

File: lib/dynamode/storage/index.ts (hunk 1 above)
Comment (blazejkustra) [reply lib/dynamode/storage/index.ts:168]: This was done to ensure that decorated keys from the entity matches these defined in Table's metadata (second argument of `TableManager` constructor). 

The only valid role for primary partition key is 'partitionKey', that's why it's provided

**Comment by blazejkustra** [comment]
> One remark: as in the proposed example, the original case from #30 looks good.
> 
> However, one of the allowed scenarios does not work:
> 
> ```ts
> export class TableUser extends Entity {
>   @attribute.partitionKey.string()
>   id: string;
> 
>   // --> This throws a "role mismatch" - and it's perfectly allowed as a reprojection for GSI. <--
>   // "Attribute "type" is decorated with a wrong role in "TableUser" Entity."
>   @attribute.sortKey.string()
>   @attribute.gsi.partitionKey.string({ indexName: 'index-1' })
>   type: string;
> 
>   @attribute.gsi.sortKey.string({ indexName: 'index-1' })
>   email: string;
>   
>   // ...
> }
> ```

Should be fixed with https://github.com/blazejkustra/dynamode/pull/31/commits/3edd9548bf2be21d1be998793c3d2ba874e80c69, great find! (I haven't used DynamoDB for a while so I forgot about this use case which is 100% valid)

File: lib/dynamode/storage/index.ts (hunk 1 above)
Comment (afronski) [reply lib/dynamode/storage/index.ts:168]: Oh yeah, indeed! In this instance, it makes perfect sense - as in the latest commit, later in the code, the GSI part missed the additional "role" in validation.

**Comment by afronski** [comment]
@blazejkustra Thanks for the update - I rechecked the case presented above, and it works as expected now.

_LGTM_. 🙇

**Comment by blazejkustra** [comment]
I'll add some e2e tests and merge this week. Thank you!

**Comment by blazejkustra** [comment]
Update: I found one issue connected to entity metadata caused by this change 😢 I have to rethink how `getEntityAttributes` is implemented but I have little time this week.

**Comment by blazejkustra** [comment]
Turned out it is an edge case that I can fix in a different PR. Added remaining tests and merging now. Thanks @afronski for the review!
